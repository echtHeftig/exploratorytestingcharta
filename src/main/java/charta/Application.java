package charta;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

import java.util.Calendar;

@SpringBootApplication
public class Application implements CommandLineRunner {

    @Autowired
    private ChartaRepository chartaRepository;

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @Override
    public void run(String... args) throws Exception {
        chartaRepository.deleteAll();

        chartaRepository.save(new Charta("Charter Name mit RandomId", "Dies sind versch. Areas", new Calendar.Builder().setDate(2018, 7, 3).build(),
                "Susanna", "", 13, 25, 25, 40,
                10, "It/is/a/path", "Dies ist eine Notiz", "Es gab noch ein paar weitere Moeglichkeiten",
                "Dies ist ein Bug. Und noch ein Bug", "Issue 1, Noch ein Issue 2"));
        chartaRepository.save(new Charta("Charter Zwei", "2Dies sind versch. Areas", new Calendar.Builder().setDate(2019, 7, 3).build(),
                "Susanna", "", 13, 25, 25, 40,
                10, "It/is/a/path", "Dies ist eine Notiz", "Es gab noch ein paar weitere Moeglichkeiten",
                "Dies ist ein Bug. Und noch ein Bug", "Issue 1, Noch ein Issue 2"));

        for(Charta charta : chartaRepository.findAll()) {
            System.out.println("####");
            System.out.println(charta.getCharterName());
            System.out.println("$$$$");
        }
    }
}
