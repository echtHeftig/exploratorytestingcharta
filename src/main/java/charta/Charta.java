package charta;

import java.util.Calendar;

public class Charta {

    private final long id;
    private String charterName;
    private String areas;
    private Calendar start;
    private String nameOfTester;
    private String taskBreakDown;
    private Integer duration;
    private Integer testDesignAndExecutionTimeInPercent;
    private Integer bugInvestigationAndReportingTimeInPercent;
    private Integer sessionSetupTimeInPercentage;
    private Integer charterVsOpportunityTimeInPercentage;
    private String dataFilesPaths;
    private String testNotes;
    private String opportunities;
    private String bugs;
    private String issues;

    public Charta(long id, String charterName, String areas, Calendar start, String nameOfTester, String taskBreakDown, Integer duration, Integer testDesignAndExecutionTimeInPercent, Integer bugInvestigationAndReportingTimeInPercent, Integer sessionSetupTimeInPercentage, Integer charterVsOpportunityTimeInPercentage, String dataFilesPaths, String testNotes, String opportunities, String bugs, String issues) {
        this.id = id;
        this.charterName = charterName;
        this.areas = areas;
        this.start = start;
        this.nameOfTester = nameOfTester;
        this.taskBreakDown = taskBreakDown;
        this.duration = duration;
        this.testDesignAndExecutionTimeInPercent = testDesignAndExecutionTimeInPercent;
        this.bugInvestigationAndReportingTimeInPercent = bugInvestigationAndReportingTimeInPercent;
        this.sessionSetupTimeInPercentage = sessionSetupTimeInPercentage;
        this.charterVsOpportunityTimeInPercentage = charterVsOpportunityTimeInPercentage;
        this.dataFilesPaths = dataFilesPaths;
        this.testNotes = testNotes;
        this.opportunities = opportunities;
        this.bugs = bugs;
        this.issues = issues;
    }

    public long getId() {
        return id;
    }
    public String getCharterName() {
        return charterName;
    }

    public void setCharterName(String charterName) {
        this.charterName = charterName;
    }

    public String getAreas() {
        return areas;
    }

    public void setAreas(String areas) {
        this.areas = areas;
    }

    public Calendar getStart() {
        return start;
    }

    public void setStart(Calendar start) {
        this.start = start;
    }

    public String getNameOfTester() {
        return nameOfTester;
    }

    public void setNameOfTester(String nameOfTester) {
        this.nameOfTester = nameOfTester;
    }

    public String getTaskBreakDown() {
        return taskBreakDown;
    }

    public void setTaskBreakDown(String taskBreakDown) {
        this.taskBreakDown = taskBreakDown;
    }

    public Integer getDuration() {
        return duration;
    }

    public void setDuration(Integer duration) {
        this.duration = duration;
    }

    public Integer getTestDesignAndExecutionTimeInPercent() {
        return testDesignAndExecutionTimeInPercent;
    }

    public void setTestDesignAndExecutionTimeInPercent(Integer testDesignAndExecutionTimeInPercent) {
        this.testDesignAndExecutionTimeInPercent = testDesignAndExecutionTimeInPercent;
    }

    public Integer getBugInvestigationAndReportingTimeInPercent() {
        return bugInvestigationAndReportingTimeInPercent;
    }

    public void setBugInvestigationAndReportingTimeInPercent(Integer bugInvestigationAndReportingTimeInPercent) {
        this.bugInvestigationAndReportingTimeInPercent = bugInvestigationAndReportingTimeInPercent;
    }

    public Integer getSessionSetupTimeInPercentage() {
        return sessionSetupTimeInPercentage;
    }

    public void setSessionSetupTimeInPercentage(Integer sessionSetupTimeInPercentage) {
        this.sessionSetupTimeInPercentage = sessionSetupTimeInPercentage;
    }

    public Integer getCharterVsOpportunityTimeInPercentage() {
        return charterVsOpportunityTimeInPercentage;
    }

    public void setCharterVsOpportunityTimeInPercentage(Integer charterVsOpportunityTimeInPercentage) {
        this.charterVsOpportunityTimeInPercentage = charterVsOpportunityTimeInPercentage;
    }

    public String getDataFilesPaths() {
        return dataFilesPaths;
    }

    public void setDataFilesPaths(String dataFilesPaths) {
        this.dataFilesPaths = dataFilesPaths;
    }

    public String getTestNotes() {
        return testNotes;
    }

    public void setTestNotes(String testNotes) {
        this.testNotes = testNotes;
    }

    public String getOpportunities() {
        return opportunities;
    }

    public void setOpportunities(String opportunities) {
        this.opportunities = opportunities;
    }

    public String getBugs() {
        return bugs;
    }

    public void setBugs(String bugs) {
        this.bugs = bugs;
    }

    public String getIssues() {
        return issues;
    }

    public void setIssues(String issues) {
        this.issues = issues;
    }
}
