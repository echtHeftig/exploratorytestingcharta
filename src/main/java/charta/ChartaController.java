package charta;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.ArrayList;
import java.util.List;

import static org.springframework.web.bind.annotation.RequestMethod.DELETE;
import static org.springframework.web.bind.annotation.RequestMethod.GET;
import static org.springframework.web.bind.annotation.RequestMethod.POST;

@RestController
public class ChartaController {

    private List<Charta> chartaList = new ArrayList<>();

    @RequestMapping(value = "/charta", method = POST)
    public Charta createAndReturnCharta(@RequestBody Charta charta) {
        chartaList.add(charta);
        return charta;
    }

    @RequestMapping(value = "/charta", method = GET)
    public String getAllChartaNamesAsConcatString() {
        StringBuilder charterList = new StringBuilder("Charters-Names: ");
        if (chartaList.isEmpty()) {
            return "The list does not contain any Charters";
        } else {
            for (Charta charta : chartaList) {
                charterList.append("\n").append(charta.getCharterName());
            }
        }
        return charterList.toString();
    }

    @RequestMapping(value = "/charta", method = DELETE)
    public String deleteAllChartas() {
        chartaList.clear();
        return chartaList.toString();
    }
}
