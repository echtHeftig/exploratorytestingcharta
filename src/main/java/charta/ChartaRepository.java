package charta;

import org.springframework.data.mongodb.repository.MongoRepository;

public interface ChartaRepository extends MongoRepository<Charta, String> {

}
