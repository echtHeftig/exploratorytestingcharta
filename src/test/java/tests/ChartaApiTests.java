package tests;

import api.ChartaApi;
import charta.Application;
import io.restassured.RestAssured;
import io.restassured.response.Response;
import org.approvaltests.Approvals;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.web.server.LocalServerPort;
import org.springframework.test.context.junit4.SpringRunner;

import java.io.File;
import java.util.Objects;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = Application.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public class ChartaApiTests {

    @LocalServerPort
    int port;

    @Before
    public void setUp() {
        RestAssured.port = port;
        ChartaApi.deleteAllChartas();
    }

    @Test
    public void getChartaListReturnsEmptyMessageIfListIsEmpty() {
        Response response = ChartaApi.getChartaList();

        Approvals.verify(response.getBody().prettyPrint());
    }
    @Test
    public void createChartaTestReturnsCorrectPayload() {
        File file = getTestDataFile("post/PostReturnsCorrectPayload.json");
        Response response = ChartaApi.postCharta(file);

        Approvals.verify(response.getBody().prettyPrint());
    }

    @Test
    public void createChartaTestReturnsCorrectPayloadSecondTest() {
        File file = getTestDataFile("post/PostReturnsCorrectPayloadSecondTest.json");
        Response response = ChartaApi.postCharta(file);

        Approvals.verify(response.getBody().prettyPrint());
    }

    private File getTestDataFile(String pathToFile) {
        ClassLoader classLoader = getClass().getClassLoader();
        return new File(Objects.requireNonNull(classLoader.getResource(pathToFile))
                .getFile());
    }
}
