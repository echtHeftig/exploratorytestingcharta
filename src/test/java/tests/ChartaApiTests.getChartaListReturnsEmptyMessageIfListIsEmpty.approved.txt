The list does not contain any Charters